### ISSUE TITLE
Deferred Intent UPE error reporting location

### ISSUE THREAD

**Issue by frosso** [Automattic/woocommerce-payments#7691]
### Description

It looks like the dUPE is rendering payment processing errors at the top of the checkout.
Other implementations (e.g.: legacy card, legacy UPE) are rendering the same error within the context of the payment block.

Notice the difference:
<img width="330" alt="281873675-185cc39f-dc51-4665-b6b0-c37dc9ea881b" src="https://github.com/Automattic/woocommerce-payments/assets/273592/e450e2bc-51e6-453c-8d6d-b22b5b8abbb9">
<img width="542" alt="281874364-fe5a4cec-1ecc-4ed8-9d0f-71e4ec33d43a" src="https://github.com/Automattic/woocommerce-payments/assets/273592/81e2ff14-40e6-4bc7-96e3-3c8e7d826679">

### Acceptance criteria

Error notice is displayed within the context of the payment block.

### Testing instructions

- Go to checkout
- Use an error card, e.g.: `4000000000000002`
- See error appear

## Dev notes

It seems that the `onPaymentSetup` event should render the error within the `noticeContexts.PAYMENTS` section, but it doesn't seem to be the case.

https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/block-client-apis/checkout/checkout-flow-and-events.md#onpaymentsetup

Look into the reason, and otherwise set the correct `messageContext` attribute.

**Comment by pierorocca** [comment]
Confirmed this is live in production and that the Design spec required the error notice to be located above the card component not at the top of the page.

**Comment by nikivias** [comment]
@frosso Is there anything needed from me for this issue? Otherwise, I agree it should be included within the payment options. [See the Figma mockups](https://www.figma.com/design/rCgVEfu8oC5NBGUzWxVvuw/WooPay-2.0?node-id=3311-418862&t=GCUX2XQvmVQtqcFz-1)

**Comment by frosso** [comment]
@nikkivias I don't think anything is needed from your end - it seems like we should just ensure the notice is displayed within the payment method options (rather than the top of checkout).

**Comment by timur27** [comment]
Quick update on this issue: deferred intent creation UPE behaves similarly to how legacy credit card behaved in the past, which is visible on the screenshots in the description. There is certain technical aspect in Blocks that I _currently_ consider a limitation on how the errors returned from the checkout API can be handled. I asked Blocks team about it in p1718284000418279-slack-C02TS23QJ1X and I'll follow this up.

**Comment by timur27** [comment]
I updated the issue estimation from `1SP` to `3SP` because it turned out to be a bit more complex than adding error message context within the existing `onPaymentSetup` event observer. The error occurred at a later stage of payment processing. We had to modify the error handling flow in our main gateway, debug the Core & Blocks flows, and handle checkout processor errors using the newly introduced `onCheckoutFail` event observer.
